hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/protocolPB/OzoneManagerProtocolClientSideTranslatorPB.java

```diff
@@ -1461,6 +1462,30 @@ public ServiceInfoEx getServiceInfo() throws IOException {
         resp.getCaCertificate(), resp.getCaCertsList());
   }
 
+  @Override
+  public void transferLeadership(String host, boolean isRandom)
+      throws IOException {
+    TransferOmLeadershipRequest.Builder builder = TransferOmLeadershipRequest.
+        newBuilder();
+    if (isRandom) {
+      builder.setIsRandom(true);
+    } else {
+      String pattern = "^((2((5[0-5])|([0-4]\\d)))|([0-1]?\\d{1,2}))(\\." +
+          "((2((5[0-5])|([0-4]\\d)))|([0-1]?\\d{1,2}))){3}:\\d+$";
+      if (host == null || !host.matches(pattern)) {
+        throw new IllegalArgumentException("Host is " + host + " or does not " +
```

Comment:
This piece of code is duplicated in SCM and OM.  I think it makes sense to extract it in a shared class

Reply:
Thx for the idea. Will update it